File: objc/iphone/NSOperationSample/NSOperationSample/NestedOperation.m
//
//  NestedOperation.m
//  NSOperationSample
//
//  Created by Taisuke Fujita on 11/05/26.
//  Copyright 2011 Commonsense. All rights reserved.
//

#import "NestedOperation.h"


@implementation NestedOperation

- (BOOL)isConcurrent {
    return NO;
}

- (void)main {
    NSLog(@"%@", @"NestedOperation!!!");
    NSLog(@"isMainThread:%d", [NSThread isMainThread]);
    NSLog(@"Thread:%@", [NSThread currentThread]);
}

@end

File: ios_swift/NSOperation/NSOperation/NestedOperation.swift
//
//  NormalOperation.swift
//  NSOperation
//
//  Created by taisuke fujita on 2014/10/04.
//  Copyright (c) 2014年 taisuke fujita. All rights reserved.
//

import UIKit

class NestedOperation: NSOperation {
    override func main() {
        NSLog("Nested isMainThread:\(NSThread.isMainThread())")
        NSLog("Nested Thread:\(NSThread.currentThread())")
    }
}
